bee-network/src/init.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#![cfg(feature = "full")]
+
+use super::{
+    config::NetworkConfig,
+    peer::list::{PeerList, PeerListWrapper},
+    service::{
+        command::command_channel,
+        controller::{NetworkCommandSender, NetworkEventReceiver},
+        event::{event_channel, Event, InternalEvent},
+    },
+    types::{PeerInfo, PeerRelation},
+    Keypair, PeerId,
+};
+
+use crate::{
+    alias,
+    network::host::NetworkHostConfig,
+    service::service::{self, NetworkServiceConfig},
+};
+
+use libp2p::identity;
+use log::info;
+use once_cell::sync::OnceCell;
+
+pub mod global {
+    use super::*;
+
+    static RECONNECT_INTERVAL_SECS: OnceCell<u64> = OnceCell::new();
+    static NETWORK_ID: OnceCell<u64> = OnceCell::new();
+    static MAX_UNKNOWN_PEERS: OnceCell<usize> = OnceCell::new();
+
+    pub fn set_reconnect_interval_secs(reconnect_interval_secs: u64) {
+        if cfg!(test) {
+            let _ = RECONNECT_INTERVAL_SECS.set(reconnect_interval_secs);
+        } else {
+            RECONNECT_INTERVAL_SECS
+                .set(reconnect_interval_secs)
+                .expect("oncecell set");
+        }
+    }
+    pub fn reconnect_interval_secs() -> u64 {
+        *RECONNECT_INTERVAL_SECS.get().expect("oncecell get")
+    }
+
+    pub fn set_network_id(network_id: u64) {
+        if cfg!(test) {
+            let _ = NETWORK_ID.set(network_id);
+        } else {
+            NETWORK_ID.set(network_id).expect("oncecell set");
+        }
+    }
+    pub fn network_id() -> u64 {
+        *NETWORK_ID.get().expect("oncecell get")
+    }
+
+    pub fn set_max_unknown_peers(max_unknown_peers: usize) {
+        if cfg!(test) {
+            let _ = MAX_UNKNOWN_PEERS.set(max_unknown_peers);
+        } else {
+            MAX_UNKNOWN_PEERS.set(max_unknown_peers).expect("oncecell set");
+        }
+    }
+    pub fn max_unknown_peers() -> usize {
+        *MAX_UNKNOWN_PEERS.get().expect("oncecell get")
+    }
+}
+
+/// Initializes a "standalone" version of the network layer.
+pub mod standalone {
+    use super::*;
+    use crate::{network::host::standalone::NetworkHost, service::service::standalone::NetworkService};
+
+    use futures::channel::oneshot;
+
+    use std::future::Future;
+
+    /// Initializes the network.
+    pub async fn init(
+        config: NetworkConfig,
+        keys: Keypair,
+        network_id: u64,
+        shutdown: Box<dyn Future<Output = ()> + Send + Unpin>,
+    ) -> (NetworkCommandSender, NetworkEventReceiver) {
+        let (network_config, service_config, network_command_sender, network_event_receiver) =
+            __init(config, keys, network_id);
```

Comment:
What's the reason for this double-underscore name? This is only usually useful in a language without namespaces.

Reply:
artistic license :smile: